components/calendar/Cargo.toml

```diff
@@ -33,6 +33,8 @@ databake = { workspace = true, features = ["derive"], optional = true }
 serde = { workspace = true, features = ["derive", "alloc"], optional = true }
 
 icu_calendar_data = { workspace = true, optional = true }
+icu_provider_baked = { workspace = true, optional = true }
+icu_locale = { workspace = true, optional = true, features = ["compiled_data"] }
```

Comment:
Nit, here and elsewhere:

```toml
icu_locale = { workspace = true, optional = true }

compiled_data = ["dep:icu_calendar_data", "dep:icu_provider_baked", "dep:icu_locale", "icu_locale?/compiled_data"]
```

Reply:
I'll make the change, but what's the reason for doing this?

(Also the `?` isn't necessary but we can be explicit I guess)

Reply:
1. When reading Cargo.toml files, I don't want to have to think about whether the `icu_locale` dependency is because of baked data (in which case the always-on `compiled_data` feature is ok) or because some other feature of the library needs it (in which case the feature should not be enabled)
2. It's cleaner for features of the same name to be enabled by other features of that name in other Cargo.toml files. So the place where we turn on the `compiled_data` feature of a dependency crate should be in the `compiled_data` feature of the dependent crate.

About the `?`, I don't care as much about this except that we have precedent elsewhere of being explicit about the `?`. I remember this has come up before.

Reply:
That makes sense

I think the antipattern for `?` is having `foo/data` without having `foo` (because it's implicit). I'd actually like there to be a clippy lint for it. On the other hand I think `foo` + `foo?/data` is unnecessary, we can just have `foo` + `foo/data`. So I wouldn't _fix_ these cases, but I was touching it anyway and I didn't want to have another cycle of discussion. So I think we shouldn't be requiring those changes in the future: I don't think we've set a precedent about using `?` when the crate is already being enabled.